### PR TITLE
fix(igx-tabs): check for undefined _resizeObserver in ngOnDestroy

### LINKED ISSUE #9765: IgxTabs not rendering with dynamic/asynchronous router-outlet
## Description  
This bug seems to be related to the one already fixed in the [DataGrid](https://github.com/IgniteUI/igniteui-angular/issues/8071).

Quick summary: The `IgxTabs` do not render when integrated into a page that gets asynchronously/dynamically rendered in a `router-outlet`.

This [respository](https://github.com/remshams/igx-tabs-issue) showcases the issue. After clicking on the login button the error added to the Attachments section occurs.

 * igniteui-angular version: 12.0.4
 * browser: Chrome (91.0.4472.106), Firefox (89.0.1)

I have opened a [PR](https://github.com/IgniteUI/igniteui-angular/pull/9766) which should fix the issue. Please let me know if I should change something or if you prefer to do the changes yourself

## Steps to reproduce  

1. Start the example application from the master branch
2. Open the DevConsole in Chrome/Firefox
3. Click on "Login"

## Result  
The page does not render and instead an error is shown in the console.

The reasons seems that `ngOnDestroy` is called but `ngAfterViewInit` is not. For that reason `_resizeObserver` not being assigned/setup properly leading to the error shown below.

## Expected result  
Page/Tabs should render.

## Attachments  
Attach a sample if available, and screenshots, if applicable.  
igx-tabs-disconnect

### PR DESCRIPTION
**Pull request by remshams**
Closes #9765

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/tabs/tabs/tab-header.component.ts [hunk 1]
```diff
@@ -115,9 +115,11 @@ export class IgxTabHeaderComponent extends IgxTabHeaderDirective implements Afte
 
     /** @hidden @internal */
     public ngOnDestroy(): void {
-        this.ngZone.runOutsideAngular(() => {
-            this._resizeObserver.disconnect();
-        });
+        if (this._resizeObserver) {
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/tabs/tabs/tab-header.component.ts:118]: Wouldn't it be simpler just to do a check before calling `disconnect()`?
```suggestion
    /** @hidden @internal */
    public ngOnDestroy(): void {
          this.ngZone.runOutsideAngular(() => {
              this._resizeObserver?.disconnect();
          });
    }

```

**Review by kdinev — CHANGES_REQUESTED** [review]

File: projects/igniteui-angular/src/lib/tabs/tabs/tab-header.component.ts (hunk 1 above)
Comment (kdinev) [reply projects/igniteui-angular/src/lib/tabs/tabs/tab-header.component.ts:118]: One more thing - `master` currently tracks `12.1.x`. If you want the fix to appear in a patch in `12.0.x` then you need to branch from `12.0.x` cherry-pick your changes and create an additional PR against `12.0.x`.

**Comment by kdinev** [comment]
`master` currently tracks `12.1.x`. If you want the fix to appear in a patch in 12.0.x then you need to branch from 12.0.x cherry-pick your changes and create an additional PR against 12.0.x.

File: projects/igniteui-angular/src/lib/tabs/tabs/tab-header.component.ts [hunk 2]
```diff
@@ -115,9 +115,11 @@ export class IgxTabHeaderComponent extends IgxTabHeaderDirective implements Afte
...
     /** @hidden @internal */
     public ngOnDestroy(): void {
-        this.ngZone.runOutsideAngular(() => {
-            this._resizeObserver.disconnect();
-        });
+        if (this._resizeObserver) {
+            this.ngZone.runOutsideAngular(() => {
+                this._resizeObserver.disconnect();
+            });
+        }
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/tabs/tabs/tab-header.component.ts:122]:
```suggestion
        this.ngZone.runOutsideAngular(() => {
            this._resizeObserver?.disconnect();
        });
```

**Review by kdinev — CHANGES_REQUESTED** [review]

File: projects/igniteui-angular/src/lib/tabs/tabs/tab-header.component.ts (hunk 2 above)
Comment (remshams) [reply projects/igniteui-angular/src/lib/tabs/tabs/tab-header.component.ts:122]: My idea was to completely avoid the `runOutsideAngular` call (as the `disconnect` is the only thing being done there).

What you suggested should do the trick as well, so I can change it to your version if your prefer it.

Comment (kdinev) [reply projects/igniteui-angular/src/lib/tabs/tabs/tab-header.component.ts:122]: I don't think the empty callback would affect performance or anything else, but the code looks cleaner this way.

Comment (remshams) [reply projects/igniteui-angular/src/lib/tabs/tabs/tab-header.component.ts:122]: Good point, thanks for your feedback. I will change it to your suggestion.

**Comment by remshams** [comment]
> `master` currently tracks `12.1.x`. If you want the fix to appear in a patch in `12.0.x` then you need to branch from `12.0.x`, cherry-pick your changes and create an additional PR against `12.0.x`.

Thanks for the info. Is it okay if I create the PR when this one is approved?

**Review by kdinev — APPROVED** [review]

**Comment by kdinev** [comment]
@remshams I will be merging this PR. Please cherry-pick the changes and create a new PR against `12.0.x`

**Pull request merged**